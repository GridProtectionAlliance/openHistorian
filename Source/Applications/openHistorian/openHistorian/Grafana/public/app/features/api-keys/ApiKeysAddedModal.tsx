import React from 'react';
import { Icon } from '@grafana/ui';

export interface Props {
  apiKey: string;
  rootPath: string;
}

export const ApiKeysAddedModal = (props: Props) => {
  return (
    <div className="modal-body">
      <div className="modal-header">
        <h2 className="modal-header-title">
          <Icon name="key-skeleton-alt" size="lg" />
          <span className="p-l-1">API Key Created</span>
        </h2>

        <a className="modal-header-close" ng-click="dismiss();">
          <Icon name="times" />
        </a>
      </div>

      <div className="modal-content">
        <div className="gf-form-group">
          <div className="gf-form">
            <span className="gf-form-label">Key</span>
            <span className="gf-form-label">{props.apiKey}</span>
          </div>
        </div>

        <div className="grafana-info-box" style={{ border: 0 }}>
          You will only be able to view this key here once! It is not stored in this form. So be sure to copy it now.
          <br />
          <br />
          You can authenticate request using the Authorization HTTP header, example:
          <br />
          <br />
          <pre className="small">
            curl -H "Authorization: Bearer {props.apiKey}" {props.rootPath}/api/dashboards/home
          </pre>
        </div>
      </div>
    </div>
  );
};

export default ApiKeysAddedModal;
