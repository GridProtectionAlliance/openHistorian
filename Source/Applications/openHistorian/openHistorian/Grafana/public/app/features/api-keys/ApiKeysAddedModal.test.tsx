import { render, screen } from '@testing-library/react';
import React from 'react';

import { ApiKeysAddedModal, Props } from './ApiKeysAddedModal';

describe('ApiKeysAddedModal', () => {
  const props: Props = {
    onDismiss: jest.fn(),
    apiKey: 'myApiKey',
    rootPath: 'test/path',
  };

  it('should render without throwing', () => {
    expect(() => render(<ApiKeysAddedModal {...props} />)).not.toThrow();
  });

  it('displays the apiKey in a readOnly input', () => {
    render(<ApiKeysAddedModal {...props} />);
    const input = screen.getByRole('textbox');
    expect(input).toHaveValue(props.apiKey);
    expect(input).toHaveAttribute('readonly');
  });

  it('has a `Copy to clipboard` button', () => {
    render(<ApiKeysAddedModal {...props} />);
    expect(screen.getByRole('button', { name: 'Copy' })).toBeInTheDocument();
  });

  it('displays the correct curl path', () => {
    render(<ApiKeysAddedModal {...props} />);
    expect(
      screen.getByText('curl -H "Authorization: Bearer myApiKey" test/path/api/dashboards/home')
    ).toBeInTheDocument();
  });

  it('calls onDismiss when the modal is closed', () => {
    render(<ApiKeysAddedModal {...props} />);
    screen.getByRole('button', { name: 'Close dialogue' }).click();
    expect(props.onDismiss).toHaveBeenCalled();
  });
});
