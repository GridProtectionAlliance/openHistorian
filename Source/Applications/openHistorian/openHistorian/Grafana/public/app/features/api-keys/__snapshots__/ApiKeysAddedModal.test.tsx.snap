// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="modal-body"
>
  <div
    className="modal-header"
  >
    <h2
      className="modal-header-title"
    >
      <Icon
        name="key-skeleton-alt"
        size="lg"
      />
      <span
        className="p-l-1"
      >
        API Key Created
      </span>
    </h2>
    <a
      className="modal-header-close"
      ng-click="dismiss();"
    >
      <Icon
        name="times"
      />
    </a>
  </div>
  <div
    className="modal-content"
  >
    <div
      className="gf-form-group"
    >
      <div
        className="gf-form"
      >
        <span
          className="gf-form-label"
        >
          Key
        </span>
        <span
          className="gf-form-label"
        >
          api key test
        </span>
      </div>
    </div>
    <div
      className="grafana-info-box"
      style={
        Object {
          "border": 0,
        }
      }
    >
      You will only be able to view this key here once! It is not stored in this form. So be sure to copy it now.
      <br />
      <br />
      You can authenticate request using the Authorization HTTP header, example:
      <br />
      <br />
      <pre
        className="small"
      >
        curl -H "Authorization: Bearer 
        api key test
        " 
        test/path
        /api/dashboards/home
      </pre>
    </div>
  </div>
</div>
`;
